lib/analyzer/applications/java.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from "path";
+import { JarFingerprint } from "../types";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+function mapContentToJarFingerprints(input: {
+  [fileName: string]: string;
+}): {
+  [path: string]: JarFingerprint[];
+} {
+  const jarFingerprints: JarFingerprint[] = Object.entries(input).map(
+    ([filePath, digest]) => {
+      return {
+        location: filePath,
+        digest,
+      };
+    },
+  );
+
+  const resultAggregatedByPath = jarFingerprints.reduce(
```

Comment:
nit: for readability this could be a for-loop too! Since you're not necessarily reducing the array to a single value but mainly restructuring the data